### PR TITLE
chore: implement document screenshots in BiDi

### LINKED ISSUE #11333: [Bug]: addStyleTag does not work when screenshot
### Minimal, reproducible example

```TypeScript
const browser = await puppeteer.launch({
    // headless: true,
    // args: ['--no-sandbox']
    headless: false,
    slowMo: 250, // slow down by 250ms
  });
  const page = await browser.newPage();
  await page.emulate(phone);
  await page.goto(url);

  await page.evaluate(() => {
    document.querySelectorAll('img').forEach((item, index) => {
      if(item.dataset.src) {
        item.src = item.dataset.src;
        item.classList.remove('img_loading')
      }
    })
  })
  await autoScroll(page);

  await page.addStyleTag({
    content: `
        #activity-name{display:none!important;}
      `
  });
  await page.screenshot({
    path: originLocalPath,
    fullPage: true
  });
```

### Error string

no error

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Background

This project has 2 years ago, it use "puppeteer": "^13.5.2", everything worked.

Today, I want update, so I install puppeteer 21.5.0, and page.addStyleTag didn't worked now.

macos 13.4.1 , node v17.1.0

### Expectation

I expected get a screenshot image,without title.
'#activity-name' is title

### Reality

no css injected when screenshot.

when I set :
    headless: false,
    slowMo: 250,

I saw css injected, and the title had gone.
but when the code running to page.screenshot, the page resize,and the title is show.

So, I change page.screenshot() to page.pdf(),the pdf didn't show the title, it worked.
Just not worked with page.screenshot()

Any ideas? Thanks a lot

### Puppeteer configuration file (if used)

_No response_

### Puppeteer version

21.5.0

### Node version

v17.1.0

### Package manager

npm

### Package manager version

8.1.2

### Operating system

macOS

### PR DESCRIPTION
**Pull request by jrandolf-google**
Fixed: https://github.com/puppeteer/puppeteer/issues/11333

### CODE REVIEW

File: packages/puppeteer-core/src/bidi/Page.ts [hunk 1]
```diff
@@ -674,14 +692,12 @@ export class BidiPage extends Page {
...
       context: this.mainFrame()._id,
+      origin: captureBeyondViewport ? 'document' : 'viewport',
       format: {
         type: `image/${type}`,
-        quality: quality ? quality / 100 : undefined,
-      },
-      clip: clip && {
-        type: 'box',
-        ...clip,
+        ...(quality !== undefined ? {quality: quality / 100} : {}),
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/bidi/Page.ts:699]: The `quality` and `box` property are hard to read, please consider either the original or an alternative.

Comment (jrandolf) [reply packages/puppeteer-core/src/bidi/Page.ts:699]: This is the only way to ensure the keys are not specified if they do not exist.

File: test/TestExpectations.json [hunk 2]
```diff
@@ -1160,7 +1160,7 @@
   {
     "testIdPattern": "[screenshot.spec] Screenshots Page.screenshot should get screenshot bigger than the viewport",
     "platforms": ["darwin", "linux", "win32"],
-    "parameters": ["webDriverBiDi"],
+    "parameters": ["cdp"],
```
Comment (OrKoN) [inline test/TestExpectations.json:1169]: why does it fail for CDP now?

File: test/golden-chrome/screenshot-offscreen-clip.png
Comment (OrKoN) [inline test/golden-chrome/screenshot-offscreen-clip.png:1]: why have the goldens changed?

**Review by OrKoN — APPROVED** [review]

**Pull request merged**